Content.Server/Command/PanicButton/PanicButtonSystem.cs

```diff
@@ -0,0 +1,84 @@
+using Content.Server.Administration.Logs;
+using Content.Server.Pinpointer;
+using Content.Server.Popups;
+using Content.Server.PowerCell;
+using Content.Server.Radio.EntitySystems;
+using Content.Shared.Database;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Radio;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Command.PanicButton
+{
+    public sealed partial class PanicButtonSystem : EntitySystem
+    {
+        [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+        [Dependency] private readonly NavMapSystem _navMap = default!;
+        [Dependency] private readonly RadioSystem _radioSystem = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly PowerCellSystem _powerCell = default!;
+        [Dependency] private readonly PopupSystem _popup = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<PanicButtonComponent, UseInHandEvent>(PanicButtonOnUseInHand);
+        }
+
+        private void PanicButtonOnUseInHand(Entity<PanicButtonComponent> panicButton, ref UseInHandEvent args)
+        {
+            var component = panicButton.Comp;
+            string radioMessage;
+
+            if (!_powerCell.HasActivatableCharge(panicButton.Owner, user: args.User))
+            {
+                args.Handled = false;
+                return;
+            }
+
+            if (!_powerCell.TryGetBatteryFromSlot(panicButton.Owner, out var batteryUid, out var battery, null) &&
+                !TryComp(panicButton.Owner, out battery))
+            {
+                args.Handled = false;
+                return;
+            }
+
+            if (battery.CurrentCharge == 0)
+            {
+                _popup.PopupEntity(Loc.GetString("comp-panic-button-cell-dead-message"), args.User, panicButton.Owner);
```

Comment:
> Also the popup "comp-panic-button-cell-dead-message" is not sent if the battery is empty, please help me, I do not know how to fix it! 😢

I'm not 100% sure that's the correct way of doing it, but gas analyzer works the same way so whynot
```suggestion
                _popup.PopupEntity(Loc.GetString("comp-panic-button-cell-dead-message"), args.User, args.User);
```

Reply:
I think I'll look into everything else a bit later (not that I can help much though)